### PR TITLE
more token values moved to yySymType's ident field

### PR DESCRIPTION
**Pull request by tiancaiamao**
continue the work of https://github.com/pingcap/tidb/pull/1516

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -386,9 +297,100 @@ import (
 
 	restrict	"RESTRICT"
 	cascade		"CASCADE"
-	no		"NO"
-	action		"ACTION"
 
+%token	<ident>
```
Comment (zimulala) [inline parser/parser.y]: Add comments for the two idents?

File: parser/parser.y [hunk 2]
```diff
@@ -2290,7 +2372,12 @@ Function:
 |	FunctionCallAgg
 
 FunctionNameConflict:
-	"DATABASE" | "SCHEMA" | "IF" | "LEFT" | "REPEAT" | "CURRENT_USER" | "CURRENT_DATE" | "VERSION" | "UTC_DATE"
+	"DATABASE" | "SCHEMA" | "IF" | "LEFT" | "REPEAT" | "CURRENT_USER" | "CURRENT_DATE" 
+| "VERSION"
+	{
+		$$ = $1
+	}
+| "UTC_DATE"
```
Comment (zimulala) [inline parser/parser.y]: Put `UTC_DATE` before `VERSION` makes `UTC_DATE` like other FunctionNameConflict except `VERSION`.

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**